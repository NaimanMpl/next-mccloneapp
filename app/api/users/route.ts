import { PrismaClient } from '@prisma/client';
const prisma = new PrismaClient();

export async function GET(request: Request) {
    console.log(await prisma.users.findFirst());
    return new Response(JSON.stringify({ users: { name: 'Hello', email: 'world@world.com' } }))
}

export async function POST(request: Request) {
    return new Response(JSON.stringify({data: await request.json()}), {
        status: 201
    })
}