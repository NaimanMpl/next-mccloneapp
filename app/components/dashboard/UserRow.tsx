import { RoleEnum } from '@/app/models/role.model'
import { Badge } from '@/components/ui/badge'
import { Button } from '@/components/ui/button'
import { Dialog, DialogTrigger } from '@/components/ui/dialog'
import { DropdownMenu, DropdownMenuContent, DropdownMenuItem, DropdownMenuLabel, DropdownMenuTrigger } from '@/components/ui/dropdown-menu'
import { TableCell, TableRow } from '@/components/ui/table'
import { MoreHorizontal } from 'lucide-react'
import Image from 'next/image'
import EditUserDialog from './EditUserDialog'
import RoleBadge from './RoleBadge'

interface UserRowProps {
  id: string,
  name: string,
  email: string,
  role: RoleEnum,
  admin: boolean,
  createdAt: Date
}

const UserRow = ({ id, name, email, role, admin, createdAt }: UserRowProps) => {

  return (
    <TableRow className='cursor-pointer'>
      <TableCell>
        <div className='flex gap-4'>
          <Image className='rounded w-10' src='/default-pp.png' alt='Minecraft Clone' width='150' height='100' />
          <div>
            <p className='font-semibold'>{name}</p>
            <p className='text-muted-foreground'>{email}</p>
          </div>
        </div>
      </TableCell>
      <TableCell>
        <p>{id}</p>
      </TableCell>
      <TableCell>
        <DropdownMenu>
          <DropdownMenuTrigger asChild>
            <RoleBadge role={role} />
          </DropdownMenuTrigger>
          <DropdownMenuContent>
            <DropdownMenuItem>
              <Badge variant={role !== 'Administrateur' ? 'destructive' : 'secondary'}>
                {role === 'Joueur' ? 'Administrateur' : 'Joueur'}
              </Badge>
            </DropdownMenuItem>
          </DropdownMenuContent>
        </DropdownMenu>
      </TableCell>
      <TableCell>
        <p>{new Date(createdAt).toLocaleDateString()}</p>
      </TableCell>
      <Dialog>
        <TableCell>
          <DropdownMenu>
            <DropdownMenuTrigger asChild>
              <Button
                aria-haspopup="true"
                size="icon"
                variant="ghost"
              >
                <MoreHorizontal />
              </Button>
            </DropdownMenuTrigger>
            <DropdownMenuContent>
              <DropdownMenuLabel>Actions</DropdownMenuLabel>
                <DialogTrigger asChild>
                  <DropdownMenuItem>
                    Ã‰diter
                  </DropdownMenuItem>
                </DialogTrigger>
              <DropdownMenuItem className='flex gap-2'>
                <span className='text-destructive'>Supprimer</span>
              </DropdownMenuItem>
            </DropdownMenuContent>
          </DropdownMenu>
          <EditUserDialog id={id} userRole={role} name={name} email={email} admin={admin} />
        </TableCell>
      </Dialog>
    </TableRow>
  )
}

export default UserRow