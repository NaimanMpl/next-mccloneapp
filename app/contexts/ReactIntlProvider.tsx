'use client';
import { ReactNode } from 'react';
import { IntlProvider } from 'react-intl';

const ReactIntlProvider = ({ children }: { children: ReactNode }) => {
  return <IntlProvider locale='fr'>{children}</IntlProvider>;
};

export default ReactIntlProvider;
