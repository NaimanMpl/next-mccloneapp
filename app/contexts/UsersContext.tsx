import { createContext, Dispatch, ReactNode, SetStateAction, useContext, useEffect, useState } from "react";
import { User } from "../models/user.model";
import { getUsers } from "../services/userservice";

interface IUsersContext {
  users: User[],
  loading: boolean,
  setUsers: Dispatch<SetStateAction<User[]>>
}

export const UsersContext = createContext<IUsersContext | undefined>(undefined);

export const UsersProvider = ({ children }: { children: ReactNode }) => {
  const [ users, setUsers ] = useState<User[]>([]);
  const [ loading, setLoading ] = useState(false);

  useEffect(() => {
    const fetchUsers = async () => {
      const users = await getUsers();
      setUsers(users);
      setLoading(false);
    }

    setLoading(true);
    fetchUsers();
  }, []);

  return (
    <UsersContext.Provider value={{ users, loading, setUsers }}>
      {children}
    </UsersContext.Provider>
  )
}

export const useUsers = () => {
  const context = useContext(UsersContext);
  if (context === undefined) {
    throw new Error("useUsers must be used within a UsersProvider");
  }
  return context;
}