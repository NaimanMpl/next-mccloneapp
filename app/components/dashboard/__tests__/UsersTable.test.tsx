import { USERS_MOCK } from "@/__mocks__/users";
import { UsersProvider } from "@/app/contexts/UsersContext";
import { RoleEnum } from "@/app/models/role.model";
import { render, screen, waitFor } from "@testing-library/react";
import user from '@testing-library/user-event';
import mockAxios from 'jest-mock-axios';
import UserRow from "../UserRow";
import UsersTable from "../UsersTable";

jest.mock('axios');

describe('<UsersTable />', () => {
  
  afterEach(() => {
    mockAxios.reset();
  });

  it('should render successfully', async () => {

    mockAxios.get.mockResolvedValueOnce({ data: USERS_MOCK });

    const users = render(
      <UsersProvider>
        <UsersTable />
      </UsersProvider>
    );

    await waitFor(() => expect(screen.getByText('John Doe')).toBeInTheDocument());

    expect(users).toMatchSnapshot();
  });
 
  it('renders error message on API failure', async () => {

    mockAxios.get.mockRejectedValueOnce(new Error('Failed to fetch users'));

    render(
      <UsersProvider>
        <UsersTable />
      </UsersProvider>
    );

    await waitFor(() => {
      expect(screen.getByText('Aucun utilisateur inscrit...')).toBeInTheDocument();
    });

  });

  it('should disable delete button if user is admin', async () => {

    mockAxios.get.mockResolvedValueOnce({ data: USERS_MOCK });
    const userMock = USERS_MOCK[0];

    render(
      <UsersProvider>
        <UserRow 
          key={userMock.id} 
          id={userMock.id} 
          name={userMock.name} 
          email={userMock.email} 
          admin={userMock.admin} 
          createdAt={userMock.createdAt}
          role={userMock.role.name as RoleEnum}
          profileIconUrl={userMock.profileIconUrl}
        />
      </UsersProvider>
    );

    const dropdownMenuButton = screen.getByRole('button');
    await user.click(dropdownMenuButton);
    const deleteButton = screen.getByText('Supprimer');
    expect(deleteButton.getAttribute('aria-disabled')).toBe('true');
  });

  it("should NOT disable delete button if user isn't admin", async () => {

    mockAxios.get.mockResolvedValueOnce({ data: USERS_MOCK });
    const userMock = USERS_MOCK[0];

    render(
      <UsersProvider>
        <UserRow 
          key={userMock.id} 
          id={userMock.id} 
          name={userMock.name} 
          email={userMock.email} 
          admin={false} 
          createdAt={userMock.createdAt}
          role={userMock.role.name as RoleEnum}
          profileIconUrl={userMock.profileIconUrl}
        />
      </UsersProvider>
    );

    const dropdownMenuButton = screen.getByRole('button');
    await user.click(dropdownMenuButton);
    const deleteButton = screen.getByText('Supprimer');
    expect(deleteButton.getAttribute('aria-disabled')).toBe(null);
  });

});