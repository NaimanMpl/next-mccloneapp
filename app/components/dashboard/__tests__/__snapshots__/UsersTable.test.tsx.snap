// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsersTable /> should render successfully 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mt-6"
      >
        <div
          class="rounded-lg border border-border bg-card text-card-foreground shadow-sm"
        >
          <div
            class="flex flex-col space-y-1.5 p-6"
          >
            <h3
              class="text-2xl font-semibold leading-none tracking-tight"
            >
              Utilisateurs
            </h3>
            <p
              class="text-sm text-muted-foreground"
            >
              Lorem ipsum, dolor sit amet consectetur adipisicing elit. Rerum, odio?
            </p>
          </div>
          <div
            class="p-6 pt-0"
          >
            <div
              class="flex justify-end gap-2 pb-4"
            >
              <button
                aria-controls="radix-:r0:"
                aria-expanded="false"
                aria-haspopup="dialog"
                class="items-center justify-center whitespace-nowrap rounded-md text-sm font-semibold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2 mr-auto flex gap-2"
                data-state="closed"
                type="button"
              >
                <svg
                  class="lucide lucide-plus w-5 h-5"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 12h14"
                  />
                  <path
                    d="M12 5v14"
                  />
                </svg>
                Ajouter un utilisateur
              </button>
              <div
                class="relative lg:w-96"
              >
                <svg
                  class="lucide lucide-search absolute left-2.5 top-2.5 h-4 w-4 text-muted-foreground"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="11"
                    cy="11"
                    r="8"
                  />
                  <path
                    d="m21 21-4.3-4.3"
                  />
                </svg>
                <input
                  class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 pl-8"
                  placeholder="Rechercher"
                />
              </div>
              <button
                aria-expanded="false"
                aria-haspopup="menu"
                class="inline-flex items-center justify-center whitespace-nowrap font-semibold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3 gap-1 text-sm"
                data-state="closed"
                id="radix-:r3:"
                type="button"
              >
                <svg
                  class="lucide lucide-list-filter h-3.5 w-3.5"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3 6h18"
                  />
                  <path
                    d="M7 12h10"
                  />
                  <path
                    d="M10 18h4"
                  />
                </svg>
                <span>
                  Filtrer
                </span>
              </button>
            </div>
            <div
              class="relative w-full overflow-auto"
            >
              <table
                class="w-full caption-bottom text-sm"
              >
                <thead
                  class="[&_tr]:border-b"
                >
                  <tr
                    class="border-b border-border transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                  >
                    <th
                      class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
                    >
                      Utilisateur
                    </th>
                    <th
                      class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
                    >
                      ID
                    </th>
                    <th
                      class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
                    >
                      Role
                    </th>
                    <th
                      class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
                    >
                      Date d'inscription
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="[&_tr:last-child]:border-0"
                >
                  <tr
                    class="border-b border-border transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted cursor-pointer"
                  >
                    <td
                      class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
                    >
                      <div
                        class="flex gap-4"
                      >
                        <img
                          alt="John Doe"
                          class="w-10 h-10 rounded"
                          src="someawesomeprofileicon.com"
                        />
                        <div>
                          <p
                            class="font-semibold"
                          >
                            John Doe
                          </p>
                          <p
                            class="text-muted-foreground"
                          >
                            johndoe@domain.com
                          </p>
                        </div>
                      </div>
                    </td>
                    <td
                      class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
                    >
                      <p>
                        1
                      </p>
                    </td>
                    <td
                      class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
                    >
                      <div
                        class="rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent bg-secondary text-secondary-foreground hover:bg-secondary/80 gap-2 flex items-center w-fit"
                      >
                        <div
                          class="w-2 h-2 bg-neutral-50 rounded-full"
                        />
                        Joueur
                      </div>
                    </td>
                    <td
                      class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
                    >
                      <p>
                        07/05/2024
                      </p>
                    </td>
                    <td
                      class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
                    >
                      <button
                        aria-expanded="false"
                        aria-haspopup="true"
                        class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-semibold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 w-10"
                        data-state="closed"
                        id="radix-:r7:"
                        type="button"
                      >
                        <svg
                          class="lucide lucide-ellipsis"
                          fill="none"
                          height="24"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                          viewBox="0 0 24 24"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <circle
                            cx="12"
                            cy="12"
                            r="1"
                          />
                          <circle
                            cx="19"
                            cy="12"
                            r="1"
                          />
                          <circle
                            cx="5"
                            cy="12"
                            r="1"
                          />
                        </svg>
                      </button>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
          <div
            class="flex items-center p-6 pt-0"
          >
            <nav
              aria-label="pagination"
              class="mx-auto flex w-full justify-center"
              role="navigation"
            >
              <ul
                class="flex flex-row items-center gap-1"
              >
                <li
                  class=""
                >
                  <a
                    aria-label="Go to previous page"
                    class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-semibold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 gap-1 pl-2.5"
                    href="#"
                  >
                    <svg
                      class="lucide lucide-chevron-left h-4 w-4"
                      fill="none"
                      height="24"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="m15 18-6-6 6-6"
                      />
                    </svg>
                    <span>
                      Précédent
                    </span>
                  </a>
                </li>
                <li
                  class=""
                >
                  <a
                    aria-current="page"
                    class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-semibold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 w-10"
                    href="#"
                  >
                    1
                  </a>
                </li>
                <li
                  class=""
                >
                  <a
                    class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-semibold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 w-10 cursor-pointer"
                  >
                    2
                  </a>
                </li>
                <li
                  class=""
                >
                  <a
                    class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-semibold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 w-10 cursor-pointer"
                  >
                    3
                  </a>
                </li>
                <li
                  class=""
                >
                  <span
                    aria-hidden="true"
                    class="flex h-9 w-9 items-center justify-center"
                  >
                    <svg
                      class="lucide lucide-ellipsis h-4 w-4"
                      fill="none"
                      height="24"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <circle
                        cx="12"
                        cy="12"
                        r="1"
                      />
                      <circle
                        cx="19"
                        cy="12"
                        r="1"
                      />
                      <circle
                        cx="5"
                        cy="12"
                        r="1"
                      />
                    </svg>
                    <span
                      class="sr-only"
                    >
                      More pages
                    </span>
                  </span>
                </li>
                <li
                  class=""
                >
                  <a
                    aria-label="Go to next page"
                    class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-semibold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 gap-1 pr-2.5"
                    href="#"
                  >
                    <span>
                      Suivant
                    </span>
                    <svg
                      class="lucide lucide-chevron-right h-4 w-4"
                      fill="none"
                      height="24"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="m9 18 6-6-6-6"
                      />
                    </svg>
                  </a>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mt-6"
    >
      <div
        class="rounded-lg border border-border bg-card text-card-foreground shadow-sm"
      >
        <div
          class="flex flex-col space-y-1.5 p-6"
        >
          <h3
            class="text-2xl font-semibold leading-none tracking-tight"
          >
            Utilisateurs
          </h3>
          <p
            class="text-sm text-muted-foreground"
          >
            Lorem ipsum, dolor sit amet consectetur adipisicing elit. Rerum, odio?
          </p>
        </div>
        <div
          class="p-6 pt-0"
        >
          <div
            class="flex justify-end gap-2 pb-4"
          >
            <button
              aria-controls="radix-:r0:"
              aria-expanded="false"
              aria-haspopup="dialog"
              class="items-center justify-center whitespace-nowrap rounded-md text-sm font-semibold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2 mr-auto flex gap-2"
              data-state="closed"
              type="button"
            >
              <svg
                class="lucide lucide-plus w-5 h-5"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 12h14"
                />
                <path
                  d="M12 5v14"
                />
              </svg>
              Ajouter un utilisateur
            </button>
            <div
              class="relative lg:w-96"
            >
              <svg
                class="lucide lucide-search absolute left-2.5 top-2.5 h-4 w-4 text-muted-foreground"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="11"
                  cy="11"
                  r="8"
                />
                <path
                  d="m21 21-4.3-4.3"
                />
              </svg>
              <input
                class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 pl-8"
                placeholder="Rechercher"
              />
            </div>
            <button
              aria-expanded="false"
              aria-haspopup="menu"
              class="inline-flex items-center justify-center whitespace-nowrap font-semibold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3 gap-1 text-sm"
              data-state="closed"
              id="radix-:r3:"
              type="button"
            >
              <svg
                class="lucide lucide-list-filter h-3.5 w-3.5"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 6h18"
                />
                <path
                  d="M7 12h10"
                />
                <path
                  d="M10 18h4"
                />
              </svg>
              <span>
                Filtrer
              </span>
            </button>
          </div>
          <div
            class="relative w-full overflow-auto"
          >
            <table
              class="w-full caption-bottom text-sm"
            >
              <thead
                class="[&_tr]:border-b"
              >
                <tr
                  class="border-b border-border transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                >
                  <th
                    class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
                  >
                    Utilisateur
                  </th>
                  <th
                    class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
                  >
                    ID
                  </th>
                  <th
                    class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
                  >
                    Role
                  </th>
                  <th
                    class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
                  >
                    Date d'inscription
                  </th>
                </tr>
              </thead>
              <tbody
                class="[&_tr:last-child]:border-0"
              >
                <tr
                  class="border-b border-border transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted cursor-pointer"
                >
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
                  >
                    <div
                      class="flex gap-4"
                    >
                      <img
                        alt="John Doe"
                        class="w-10 h-10 rounded"
                        src="someawesomeprofileicon.com"
                      />
                      <div>
                        <p
                          class="font-semibold"
                        >
                          John Doe
                        </p>
                        <p
                          class="text-muted-foreground"
                        >
                          johndoe@domain.com
                        </p>
                      </div>
                    </div>
                  </td>
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
                  >
                    <p>
                      1
                    </p>
                  </td>
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
                  >
                    <div
                      class="rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent bg-secondary text-secondary-foreground hover:bg-secondary/80 gap-2 flex items-center w-fit"
                    >
                      <div
                        class="w-2 h-2 bg-neutral-50 rounded-full"
                      />
                      Joueur
                    </div>
                  </td>
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
                  >
                    <p>
                      07/05/2024
                    </p>
                  </td>
                  <td
                    class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
                  >
                    <button
                      aria-expanded="false"
                      aria-haspopup="true"
                      class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-semibold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 w-10"
                      data-state="closed"
                      id="radix-:r7:"
                      type="button"
                    >
                      <svg
                        class="lucide lucide-ellipsis"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <circle
                          cx="12"
                          cy="12"
                          r="1"
                        />
                        <circle
                          cx="19"
                          cy="12"
                          r="1"
                        />
                        <circle
                          cx="5"
                          cy="12"
                          r="1"
                        />
                      </svg>
                    </button>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="flex items-center p-6 pt-0"
        >
          <nav
            aria-label="pagination"
            class="mx-auto flex w-full justify-center"
            role="navigation"
          >
            <ul
              class="flex flex-row items-center gap-1"
            >
              <li
                class=""
              >
                <a
                  aria-label="Go to previous page"
                  class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-semibold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 gap-1 pl-2.5"
                  href="#"
                >
                  <svg
                    class="lucide lucide-chevron-left h-4 w-4"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m15 18-6-6 6-6"
                    />
                  </svg>
                  <span>
                    Précédent
                  </span>
                </a>
              </li>
              <li
                class=""
              >
                <a
                  aria-current="page"
                  class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-semibold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 w-10"
                  href="#"
                >
                  1
                </a>
              </li>
              <li
                class=""
              >
                <a
                  class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-semibold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 w-10 cursor-pointer"
                >
                  2
                </a>
              </li>
              <li
                class=""
              >
                <a
                  class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-semibold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 w-10 cursor-pointer"
                >
                  3
                </a>
              </li>
              <li
                class=""
              >
                <span
                  aria-hidden="true"
                  class="flex h-9 w-9 items-center justify-center"
                >
                  <svg
                    class="lucide lucide-ellipsis h-4 w-4"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      r="1"
                    />
                    <circle
                      cx="19"
                      cy="12"
                      r="1"
                    />
                    <circle
                      cx="5"
                      cy="12"
                      r="1"
                    />
                  </svg>
                  <span
                    class="sr-only"
                  >
                    More pages
                  </span>
                </span>
              </li>
              <li
                class=""
              >
                <a
                  aria-label="Go to next page"
                  class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-semibold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 gap-1 pr-2.5"
                  href="#"
                >
                  <span>
                    Suivant
                  </span>
                  <svg
                    class="lucide lucide-chevron-right h-4 w-4"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m9 18 6-6-6-6"
                    />
                  </svg>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
